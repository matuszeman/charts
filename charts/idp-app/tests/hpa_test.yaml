suite: hpa
templates:
  - hpa.yaml
values:
  - values/minimum.yaml
tests:
  - it: should match snapshot with values-ci.yaml
    values:
      - ../values-ci.yaml
    asserts:
      - matchSnapshot: {}

  - it: when multi deployment
    values:
      - values/deployment.yaml
      - values/autoscaling.yaml
      - values/multi.yaml

    asserts:
      - matchSnapshot: {}
      - equal:
          path: metadata.name
          value: RELEASE-NAME-one
        documentIndex: 0
      - equal:
          path: metadata.name
          value: RELEASE-NAME-two
        documentIndex: 1
  - it: when default values are used
    values:
      - values/deployment.yaml
      - values/autoscaling.yaml
    asserts:
      - isSubset:
          path: spec.behavior
          content:
            scaleDown:
              policies:
                - periodSeconds: 60
                  type: Percent
                  value: 5
              stabilizationWindowSeconds: 600
            scaleUp:
              policies:
                - periodSeconds: 60
                  type: Percent
                  value: 5
                - periodSeconds: 60
                  type: Pods
                  value: 1
              selectPolicy: Max
              stabilizationWindowSeconds: 0

  - it: when scaleDown is disabled
    values:
      - values/deployment.yaml
      - values/autoscaling.yaml
    set:
      autoscaling:
        scaleDown:
          preset: Disabled
    asserts:
      - isSubset:
          path: spec.behavior.scaleDown
          content:
            selectPolicy: Disabled

  - it: can customize scaleDown.stabilizationWindowSeconds
    values:
      - values/deployment.yaml
      - values/autoscaling.yaml
    set:
      autoscaling:
        scaleDown:
          stabilizationWindowSeconds: 1200
    asserts:
      - equal:
          path: spec.behavior.scaleDown.stabilizationWindowSeconds
          value: 1200

  - it: when deployment.kind is StatefulSet
    values:
      - values/deployment.yaml
      - values/autoscaling.yaml
      - values/statefulset.yaml
    asserts:
      - equal:
          path: spec.scaleTargetRef.kind
          value: StatefulSet

