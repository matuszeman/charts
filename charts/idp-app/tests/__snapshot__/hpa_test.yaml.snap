should match snapshot with values-ci.yaml:
  1: |
    apiVersion: autoscaling/v2
    kind: HorizontalPodAutoscaler
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-resource-suffix
    spec:
      behavior:
        scaleDown:
          policies:
            - periodSeconds: 60
              type: Percent
              value: 5
          stabilizationWindowSeconds: 600
        scaleUp:
          policies:
            - periodSeconds: 60
              type: Percent
              value: 5
            - periodSeconds: 60
              type: Pods
              value: 1
          selectPolicy: Max
          stabilizationWindowSeconds: 0
      maxReplicas: 10
      metrics:
        - resource:
            name: cpu
            target:
              averageUtilization: 80
              type: Utilization
          type: Resource
      minReplicas: 2
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: RELEASE-NAME-resource-suffix
when multi deployment:
  1: |
    apiVersion: autoscaling/v2
    kind: HorizontalPodAutoscaler
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-one
    spec:
      behavior:
        scaleDown:
          policies:
            - periodSeconds: 60
              type: Percent
              value: 5
          stabilizationWindowSeconds: 600
        scaleUp:
          policies:
            - periodSeconds: 60
              type: Percent
              value: 5
            - periodSeconds: 60
              type: Pods
              value: 1
          selectPolicy: Max
          stabilizationWindowSeconds: 0
      maxReplicas: 10
      metrics:
        - resource:
            name: cpu
            target:
              averageUtilization: 80
              type: Utilization
          type: Resource
      minReplicas: 2
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: RELEASE-NAME-one
  2: |
    apiVersion: autoscaling/v2
    kind: HorizontalPodAutoscaler
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-two
    spec:
      behavior:
        scaleDown:
          policies:
            - periodSeconds: 60
              type: Percent
              value: 5
          stabilizationWindowSeconds: 600
        scaleUp:
          policies:
            - periodSeconds: 60
              type: Percent
              value: 5
            - periodSeconds: 60
              type: Pods
              value: 1
          selectPolicy: Max
          stabilizationWindowSeconds: 0
      maxReplicas: 10
      metrics:
        - resource:
            name: cpu
            target:
              averageUtilization: 80
              type: Utilization
          type: Resource
      minReplicas: 2
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: RELEASE-NAME-two
